GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -1046,6 +1033,33 @@ private HResult QueryFileNameHandler(string virtualPath)
             }
         }
 
+        private void MarkDirectoryAsPlaceholder(
+            string virtualPath,
+            bool isDirectory,
+            uint triggeringProcessId,
+            string triggeringProcessImageFileName)
+        {
+            string directoryPath = Path.Combine(this.Context.Enlistment.WorkingDirectoryRoot, virtualPath);
+            HResult hr = this.virtualizationInstance.MarkDirectoryAsPlaceholder(
+                directoryPath,
+                FolderContentId,
+                PlaceholderVersionId);
+
+            if (hr == HResult.Ok)
+            {
+                this.FileSystemCallbacks.OnPlaceholderFolderCreated(virtualPath, triggeringProcessImageFileName);
+            }
+            else

```

Comment:
If this fails in the "OnFolderCreated returns true" scenario, should we add the folder to ModifiedPaths.dat?

Reply:
The risk there is that if that is a top level folder and that folder gets added to the ModifiedPaths.dat, it could show all the files for that top level folder as deleted in the next status and any checkout after that could be updating a lot of files and we don't currently have a way for the user to dehydrate that folder once it gets added.